pcdet/models/dense_heads/anchor_head_single.py

```diff
@@ -1,12 +1,23 @@
+from typing import Iterable, List
+
 import numpy as np
 import torch.nn as nn
 
 from .anchor_head_template import AnchorHeadTemplate
 
 
 class AnchorHeadSingle(AnchorHeadTemplate):
```

Comment:
The class `AnchorHeadSingle` is currently used in our model, so all the proposed changes could break the current code. In particular the addition of a layer can give errors in the loading of the model, but many other problems can arise. This would impact very negatively the reproducibility of the code in my opinion and the rollback in case of problems. It also hinders the possibility of training a model without a second head, something we could want to do in the future.

For all the following reasons, I'm strongly against this breaking change.

I propose to create a completely new class (that could inherit from `AnchorHeadSingle` if needed) to add the support for multiple heads. All the following changes should support both `AnchorHeadSingle` and the new class.

I'm also wondering if [`AnchorHeadMulti`](https://github.com/promaton/OpenPCDet/blob/master/pcdet/models/dense_heads/anchor_head_multi.py) could be used for our purpose given its name 🤔

Reply:
Hmm good point, although ofcourse this change will be for all products (ortho also needs type detection I guess).
I think it may be very hard to make this feature backwards compatible tbh, I'm going to try to see what we can do but there's a bit of a balance between the value of having it backward compatible and the effort to make it so.
I'm worried it's (much) more work than just adding a new anchor head class, but let me see.

FYI AnchorHeadMulti is a class to have a separate head per class, it does not allow multiple classes per element as we do here.

Reply:
I've made it backwards compatible, you can now run the old detection model just by changing the `DENSE_HEAD.NAME` parameter in `pv-rcnn-plusplus_cfg.yaml`